src/components/app/routes/loaders/checkoutStepperLoader.ts

```diff
@@ -37,45 +47,136 @@ async function planDetailsRegisterLoader(): Promise<Response | null> {
 }
 
 /**
- * Route loader for Account Details page
+ * Route loader for Account Details page.
+ *
+ * @param {QueryClient} queryClient - TanStack Query client to ensure context and session queries.
+ * @returns {Promise<Response | null>} Redirects if prerequisites fail; otherwise null to proceed.
  */
-async function accountDetailsLoader(): Promise<Response | null> {
+async function accountDetailsLoader(queryClient: QueryClient): Promise<Response | null> {
   const authenticatedUser = getAuthenticatedUser();
   if (!authenticatedUser) {
     // If the user is NOT authenticated, redirect to PlanDetails Page.
     return redirect(CheckoutPageRoute.PlanDetails);
   }
+
+  const contextMetadata: CheckoutContextResponse = await queryClient.ensureQueryData(
+    queryBffContext(authenticatedUser?.userId || null),
+  );
+  const { fieldConstraints, pricing } = contextMetadata;
+
+  const stripePriceId = extractPriceId(pricing);
+  if (!stripePriceId) {
+    return redirect(CheckoutPageRoute.PlanDetails);
+  }
+
+  const {
+    valid,
+    invalidRoute,
+  } = await validateFormState({
+    checkoutStep: CheckoutSteps.AccountDetails,
```

Comment:
Shoud use CheckoutStep:
```suggestion
    checkoutStep: CheckoutStep.AccountDetails,
```